### DISCUSSION TITLE
Alert is empty - no data on ET HUNTING RDP Authentication Bypass Attempt

### DISCUSSION THREAD

**Discussion by IzacFrank** [Security-Onion-Solutions/securityonion#6840] [Unsupported Versions] [answered]
hello,

I have the latest version of Security Onion (2.3.91) and I get a lot of _ET HUNTING RDP Authentication Bypass Attempt_  alerts but when I drill down, I see that it is empty.  Please see the below. Do you have any idea why it happens and how can I solve it?

Thank you very much in advance.

![image](https://user-images.githubusercontent.com/93327361/149120335-430a2a70-748b-4061-ab33-1018c4248873.png)

After I drill down, it says there is no data available

![image](https://user-images.githubusercontent.com/93327361/149120460-fc30da1e-638b-4150-9bef-7dc920e8f950.png)

regards,
Isac

**Comment by dougburks** [comment]
Could be related to the space before the closing quote:

<img width="444" alt="Screen Shot 2022-01-13 at 8 21 41 AM" src="https://user-images.githubusercontent.com/1659467/149337877-8eeddb46-f261-4554-8453-a0f2cff4a4a2.png">

Once you've selected that alert, click the Hunt icon to the right of the query bar. That should give you the query in Hunt and you should be able to remove the space before the closing quote. Should look like this:
```
* AND rule.name:"ET HUNTING RDP Authentication Bypass Attempt"
```

Does that return results?

**Reply by IzacFrank** [reply]
Thank you @dougburks  There is indeed a space but still, I don't get any results after removing it in Hunt. It happened with another 4 alerts, too. I removed the space like the below but no result, either

                 >    * AND rule.name:"ET MALWARE ATTACKER IRCBot - ipconfig - PRIVMSG Command" 

![image](https://user-images.githubusercontent.com/93327361/149361612-3f6f66b9-45b8-41e2-a7a9-4e88fdb941ff.png)

Please see the below-highlighted ones, I think it is somehow related to the event type.
![image](https://user-images.githubusercontent.com/93327361/149340577-791affb2-1162-4da3-bf14-66572793cecc.png)

**Reply by dougburks** [reply]
Is this a standalone deployment or distributed deployment?

Have you tried increasing the time window of your search?

Have you tried going to Hunt and searching for something simple like `ET MALWARE` or `ET HUNTING` to see if that returns the correct results?

**Reply by Mastadamus** [reply]
This is weird but I had the same exact thing happen on the same exact alert last week. I had only one such incident of that alert and when I drilled down, nothing was present.

**Reply by Masaya-A** [reply]
Same here with _ET HUNTING RDP Authentication Bypass Attempt_ alerts.

It seems double quotes make something wrong. If I deleted them from URL, those alerts are present in the Alerts Screen.
```
https://~~~%2a%20AND%20rule.name%3A"ET%20HUNTING%20RDP%20Authentication%20Bypass%20Attempt%20"~~~
https://~~~%2a%20AND%20rule.name%3AET%20HUNTING%20RDP%20Authentication%20Bypass%20Attempt%20~~~
```

And this will happen with rules which have a space at the end of rule name.
```
alert tcp $EXTERNAL_NET any -> $HOME_NET any (msg:"ET MALWARE ATTACKER IRCBot - ipconfig - PRIVMSG Command "; flow:established,from_server; content:"PRIVMSG "; content:"ipconfig"; within:100; classtype:trojan-activity; sid:2017287; rev:4; metadata:created_at 2013_08_07, updated_at 2013_08_07;)
# alert tcp $EXTERNAL_NET any -> $HOME_NET any (msg:"ET HUNTING RDP Authentication Bypass Attempt "; flow:established,to_server; content:"|03 00 00 2f 2a|"; startswith; content:"Cookie: mstshash="; dsize:<60;  reference:url,pastebin.com/PSbQXJYL; classtype:attempted-admin; sid:2034857; rev:2; metadata:affected_product Microsoft_Terminal_Server_RDP, attack_target Server, created_at 2022_01_04, deployment Perimeter, former_category HUNTING, signature_severity Major, updated_at 2022_01_04;)
```

**Reply by IzacFrank** [reply]
@dougburks It is a standalone deployment, I tried the increase the time frame but no help.  I tried the suggestion of @Masaya-A and removed the double quotes then it worked.
![image](https://user-images.githubusercontent.com/93327361/150329484-914721f2-5ddc-4b91-96bd-f0d2e9404ca0.png)

Thank you @Masaya-A 
Thank you @dougburks

**Comment by dougburks** [comment, marked as answer]
I sent an email to Emerging Threats about the trailing space in some msg fields:
http://lists.emergingthreats.net/pipermail/emerging-sigs/2022-January/030558.html

They have replied that they will get those fixed soon.

**Reply by dougburks** [reply]
To close the loop on this, it took a couple of rounds of fixes but all ET msg fields have had trailing spaces removed:
http://lists.emergingthreats.net/pipermail/emerging-sigs/2022-January/030563.html

You can verify with the following command (it should return no output):
```
cut -d\" -f2 /opt/so/rules/nids/all.rules | sed 's| $|TRAILINGSPACE|g' |grep "TRAILINGSPACE"
```

**Reply by IzacFrank** [reply]
Yes, no output. Thanks again @dougburks
